### ISSUE TITLE
[Docs]: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy_attachment

### ISSUE THREAD

**Issue by mdcarson** [hashicorp/terraform-provider-aws#30169]
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/iam_role_policy_attachment

### Description

The first example under "Example Usage" is missing the statement block.  Without it you get unsupported block errors.

**This is how it looks now:**
```
data "aws_iam_policy_document" "assume_role" {
  effect = "Allow"

  principals {
    type        = "Service"
    identifiers = ["ec2.amazonaws.com"]
  }

  actions = ["sts:AssumeRole"]
}
```

**This is how it should look:**
```
data "aws_iam_policy_document" "assume_role" {
  statement {
    effect = "Allow"
  
    principals {
      type        = "Service"
      identifiers = ["ec2.amazonaws.com"]
    }
  
    actions = ["sts:AssumeRole"]
  }
}
```

### Would you like to implement a fix?

None

**Comment by bennylu2** [comment]
Hi @mdcarson looks like it was fixed recently in https://github.com/hashicorp/terraform-provider-aws/pull/30150 and will be released soon, thanks for reporting!

**Comment by justinretzolk** [comment]
Fixed by #30150
